sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/AppConfigurationReplicaClientsBuilder.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.config.implementation;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.policy.ExponentialBackoff;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.data.appconfiguration.ConfigurationClientBuilder;
+import com.azure.identity.ManagedIdentityCredentialBuilder;
+import com.azure.spring.cloud.config.AppConfigurationCredentialProvider;
+import com.azure.spring.cloud.config.ConfigurationClientBuilderSetup;
+import com.azure.spring.cloud.config.pipline.policies.BaseAppConfigurationPolicy;
+import com.azure.spring.cloud.config.properties.ConfigStore;
+
+public class AppConfigurationReplicaClientsBuilder implements EnvironmentAware {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppConfigurationReplicaClientsBuilder.class);
+
+    /**
+     * Invalid Connection String error message
+     */
+    public static final String NON_EMPTY_MSG = "%s property should not be null or empty in the connection string of Azure Config Service.";
+
+    /**
+     * Connection String Regex format
+     */
+    private static final String CONN_STRING_REGEXP = "Endpoint=([^;]+);Id=([^;]+);Secret=([^;]+)";
+
+    /**
+     * Invalid Formatted Connection String Error message
+     */
+    public static final String ENDPOINT_ERR_MSG = String.format("Connection string does not follow format %s.",
+        CONN_STRING_REGEXP);
+
+    private static final Pattern CONN_STRING_PATTERN = Pattern.compile(CONN_STRING_REGEXP);
+
+    private final AppConfigurationCredentialProvider tokenCredentialProvider;
+
+    private final ConfigurationClientBuilderSetup clientProvider;
+
+    private boolean isDev;
+
+    private final boolean isKeyVaultConfigured;
+
+    private final String clientId;
+
+    private final int maxRetries;
+
+    public AppConfigurationReplicaClientsBuilder(AppConfigurationCredentialProvider tokenCredentialProvider,
+        ConfigurationClientBuilderSetup clientProvider, Boolean isKeyVaultConfigured, String clientId,
+        int maxRetries) {
+        this.tokenCredentialProvider = tokenCredentialProvider;
+        this.clientProvider = clientProvider;
+        this.isKeyVaultConfigured = isKeyVaultConfigured;
+        this.clientId = clientId;
```

Comment:
This `clientId` here is weird, for it seems only used when enabling MI? But MI is another kind of TokenCredential.

Reply:
This is a holdover from pre sdk days, where the client id was passed in through config. It was never removed and still is an option.